### PR TITLE
Add plugin reader warning dialog

### PR DESCRIPTION
**Pull request by ppwadhwa**
# Description
From https://github.com/napari/napari/issues/4613, this will address # 1. 

```
1. When plugin preference IS saved, the reader dialog currently appears if there’s an error. Add an alert to the user when saving a preference will override an existing one, both in the reader and preference dialogs
```
I think I have the use case correct here, but please let me know if I'm missing something.

https://user-images.githubusercontent.com/54282105/180096067-051fd639-ba96-482d-b7de-9e675e657ebc.mov

### CODE REVIEW

**Comment by DragaDoncila** [comment]
@ppwadhwa this is indeed the use case we're talking about. It would be nice if we could avoid an additional dialog here though - I wonder if we could check the current preferences before popping the initial reader dialog, and then maybe change the text for the persistence checkbox to say something like `Override existing preference for .tif files, my-tif-plugin`?

**Comment by Carreau** [comment]
This looks good to me, can we make the improvement to merge the two dialogs to a subsequent iteration of this PR ?

**Review by Carreau — APPROVED** [review]

File: napari/_qt/dialogs/qt_reader_dialog.py [hunk 1]
```diff
@@ -250,6 +266,17 @@ def open_with_dialog_choices(
     qt_viewer.viewer.open(paths, stack=stack, plugin=plugin_name, **kwargs)
 
     if persist:
+        if (
+            extension
+            or '*.' + extension
+            in get_settings().plugins.extension2reader.keys()
+        ):
+            res = qt_viewer.children()[2]._show_persist_warning_dialog(
```
Comment (Carreau) [inline napari/_qt/dialogs/qt_reader_dialog.py]: It seem like this "[2]" is magical so we can't rely on this to get the readerdialog.

**Review by Carreau — CHANGES_REQUESTED** [review]

**Comment by DragaDoncila** [comment]
I'm extremely hesitant to merge this with the additional dialog, as the required code change to simply change the message for the persistence checkbox is very compact and much more usable.

File: napari/_qt/dialogs/qt_reader_dialog.py [hunk 2]
```diff
@@ -67,9 +67,18 @@ def setup_ui(self, error_message, readers):
...
+            if (
+                '*' + self._extension
+                in get_settings().plugins.extension2reader.keys()
+            ):
+                pref = get_settings().plugins.extension2reader[
+                    '*' + self._extension
+                ]
+                warn_message = f'Override existing preference for *{self._extension}, "{pref}"'
+            else:
+                warn_message = f'Remember this choice for files with a {self._extension} extension'
```
Comment (Czaki) [inline napari/_qt/dialogs/qt_reader_dialog.py]: Please add a translation of these strings.

Comment (ppwadhwa) [reply napari/_qt/dialogs/qt_reader_dialog.py]: thanks for catching that, @Czaki !

**Comment by ppwadhwa** [comment]
I'm not sure what happened, but need to put some things back in that I guess got taken out.  This is not working as intended right now. stay tuned...fixing it.

**Comment by ppwadhwa** [comment]
@DragaDoncila  I got this one mixed up with the folders in preferences PR.  :(. So, this one was fine.  I added in translations and it should be good to go.  Nevertheless, please double check its working for you as expected.  thanks!

File: napari/_qt/dialogs/qt_reader_dialog.py [hunk 3]
```diff
@@ -67,9 +67,25 @@ def setup_ui(self, error_message, readers):
 
         # checkbox to remember the choice (doesn't pop up for folders with no extension)
         if self._extension:
-            self.persist_checkbox = QCheckBox(
-                f'Remember this choice for files with a {self._extension} extension'
-            )
+            if (
+                '*' + self._extension
```
Comment (DragaDoncila) [inline napari/_qt/dialogs/qt_reader_dialog.py:72]: this long statement in the `if` is leading to some not great formatting here. Could we use the dictionary `get` method to just save the preference into a variable directly e.g.

```python
existing_pref = get_settings().plugins.extension2reader.get('*' + self._extension)
if existing_pref:
    warn_message = ...
```

File: napari/_qt/dialogs/qt_reader_dialog.py [hunk 4]
```diff
@@ -67,9 +67,25 @@ def setup_ui(self, error_message, readers):
...
+                in get_settings().plugins.extension2reader.keys()
+            ):
+                pref = get_settings().plugins.extension2reader[
+                    '*' + self._extension
+                ]
+                warn_message = trans._(
+                    'Override existing preference for *{extension}, "{pref}"',
```
Comment (DragaDoncila) [inline napari/_qt/dialogs/qt_reader_dialog.py]: Just minor tweak here to use similar language as the initial statement.

```suggestion
                    'Override existing preference for files with a {extension} extension: {pref}',
```

File: napari/_qt/dialogs/qt_reader_dialog.py [hunk 5]
```diff
@@ -67,9 +67,25 @@ def setup_ui(self, error_message, readers):
...
         # checkbox to remember the choice (doesn't pop up for folders with no extension)
         if self._extension:
-            self.persist_checkbox = QCheckBox(
-                f'Remember this choice for files with a {self._extension} extension'
-            )
+            if (
+                '*' + self._extension
+                in get_settings().plugins.extension2reader.keys()
+            ):
+                pref = get_settings().plugins.extension2reader[
```
Comment (DragaDoncila) [inline napari/_qt/dialogs/qt_reader_dialog.py]: the settings file stores plugin name not display name, so we should get the display name of the plugin here.

**Review by DragaDoncila — COMMENTED** [review]
@ppwadhwa love it 🎉 ! Requested three minor changes, one for code styling and two for display consistency. Let me know if I need to clarify anywhere!

**Review by Czaki — APPROVED** [review]

**Review by DragaDoncila — APPROVED** [review]
LGTM!

**Comment by Czaki** [comment]
@Carreau could you review as your request changes seem to be addressed?

**Review by Carreau — APPROVED** [review]

**Pull request merged**